plugins/metrics/epochcommittment.go

```diff
@@ -1,28 +1,168 @@
+
+	numberOfSeenOtherCommitments++
+})
+
+var onCommitmentMissing = event.NewClosure(func(commitment commitment.ID) {
+	commitmentsMutex.Lock()
+	defer commitmentsMutex.Unlock()
+
+	missingCommitments++
+})
+
+var onMissingCommitmentReceived = event.NewClosure(func(commitment commitment.ID) {
+	commitmentsMutex.Lock()
+	defer commitmentsMutex.Unlock()
+
+	missingCommitmentsReceived++
 })
 
 // LastCommittedEpoch returns the last committed epoch.
 func LastCommittedEpoch() *commitment.Commitment {
 	lastCommittedEpochMutex.RLock()
 	defer lastCommittedEpochMutex.RUnlock()
-	return lastCommittedEpoch
+
+	return lastCommitment
+}
+
+// NumberOfSeenOtherCommitments returns the number of commitments different from ours that were observed by the node.
+func NumberOfSeenOtherCommitments() int {
+	seenOtherCommitmentsMutex.RLock()
+	defer seenOtherCommitmentsMutex.RUnlock()
+
+	return numberOfSeenOtherCommitments
+}
+
+// MissingCommitmentsRequested returns the number of requested missing commitments.
+func MissingCommitmentsRequested() int {
+	commitmentsMutex.RLock()
+	defer commitmentsMutex.RUnlock()
+
+	return missingCommitments
+}
+
+// MissingCommitmentsReceived returns the number of received missing commitments.
+func MissingCommitmentsReceived() int {
+	commitmentsMutex.RLock()
+	defer commitmentsMutex.RUnlock()
+
+	return missingCommitmentsReceived
+}
+
+// BlocksOfEpoch returns the number of blocks in the current epoch.
+func BlocksOfEpoch() (ei epoch.Index, accBlocks int32, accTxs int32, activeValidators int32) {
+	epochCountsMutex.Lock()
+	defer epochCountsMutex.Unlock()
+
+	return currentEI, acceptedBlocksCount.Load(), acceptedTransactionsCount.Load(), activeValidatorsCount.Load()
+}
+
+func updateBlkOfEpoch(ei epoch.Index, accBlocks int32, accTxs int32, activeValidators int32) {
+	epochCountsMutex.Lock()
+	defer epochCountsMutex.Unlock()
+
+	currentEI = ei
+	acceptedBlocksCount.Store(accBlocks)
+	acceptedTransactionsCount.Store(accTxs)
+	activeValidatorsCount.Store(activeValidators)
+}
+
+// RemovedBlocksOfEpoch returns the number of orphaned blocks in the given epoch.
+func RemovedBlocksOfEpoch() map[epoch.Index]int {
+	orphanedBlkRemovedMutex.Lock()
+	defer orphanedBlkRemovedMutex.Unlock()
+
+	checkLimit()
+
+	// copy the original map
+	clone := make(map[epoch.Index]int)
+	endEpoch := oldestEpoch + epoch.Index(orphanedBlkRemoved.Size())
+	if endEpoch == 0 {
+		endEpoch = epoch.IndexFromTime(time.Now())
+	}
+
+	for ei := oldestEpoch; ei <= endEpoch; ei++ {
+		num, exists := orphanedBlkRemoved.Get(ei)
+		if !exists {
+			num = 0
+			orphanedBlkRemoved.Set(ei, 0)
+		}
+		clone[ei] = num
+	}
+
+	return clone
+}
+
+func increaseRemovedBlockCounter(blkID models.BlockID) {
+	orphanedBlkRemovedMutex.Lock()
+	defer orphanedBlkRemovedMutex.Unlock()
+
+	if count, exists := orphanedBlkRemoved.Get(blkID.EpochIndex); exists {
+		count++
+		orphanedBlkRemoved.Set(blkID.EpochIndex, count)
+	} else {
+		orphanedBlkRemoved.Set(blkID.EpochIndex, 1)
```

Comment:
even if it is not set the get should always return the default value (in this case 0). So the if shouldn't be necessary